geode-core/src/main/java/org/apache/geode/internal/cache/partitioned/RegionAdvisor.java

```diff
@@ -180,19 +180,23 @@ public void processProfilesQueuedDuringInitialization() {
               logger.trace(LogMarker.DISTRIBUTION_ADVISOR_VERBOSE,
                   "applying queued profile removal for all buckets for {}", qbp.memberId);
             }
-            for (int i = 0; i < buckets.length; i++) {
-              BucketAdvisor ba = buckets[i].getBucketAdvisor();
-              int serial = qbp.serials[i];
-              if (serial != ILLEGAL_SERIAL) {
-                ba.removeIdWithSerial(qbp.memberId, serial, qbp.destroyed);
-              }
-            } // for
+            if (qbp.serials.length == 1) {
+              buckets[qbp.bucketId].getBucketAdvisor().removeIdWithSerial(qbp.memberId,
+                  qbp.serials[0], qbp.destroyed);
+            } else {
+              for (int i = 0; i < buckets.length; i++) {
+                BucketAdvisor ba = buckets[i].getBucketAdvisor();
+                int serial = qbp.serials[i];
+                if (serial != ILLEGAL_SERIAL) {
+                  ba.removeIdWithSerial(qbp.memberId, serial, qbp.destroyed);
+                }
+              } // for
```

Comment:
Good place to extract this as a private method of its own. The purpose isn't to allow multiple code paths to call it, just to keep methods shorter for JIT optimization and make it more readable (if you use a good method name).

Reply:
Please create a new ticket to refactor the QueuedBucketProfile.